From 97b93b55c1abc9c56c9dcbcaaf6de27bb90eb852 Mon Sep 17 00:00:00 2001
From: phanirithvij <phanirithvij2000@gmail.com>
Date: Sun, 16 Aug 2026 22:54:27 +0530
Subject: [PATCH 2/2] Fix cookie parser to handle multiple cookies correctly

---
 src/session.rs | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/src/session.rs b/src/session.rs
index 17bc782..32e1c92 100644
--- a/src/session.rs
+++ b/src/session.rs
@@ -133,11 +133,13 @@ impl FromRequestParts<AppState> for Session {
 
 fn extract_session_key(cookie_headers: Vec<&str>) -> Option<Uuid> {
     for cookie_header in cookie_headers {
-        if let Some((name, value)) = cookie_header.split_once('=')
-            && name.trim() == COOKIE_NAME
-            && let Ok(uuid) = value.trim().parse::<Uuid>()
-        {
-            return Some(uuid);
+        for cookie in cookie_header.split(';') {
+            if let Some((name, value)) = cookie.split_once('=')
+                && name.trim() == COOKIE_NAME
+                && let Ok(uuid) = value.trim().parse::<Uuid>()
+            {
+                return Some(uuid);
+            }
         }
     }
     None
-- 
2.55.0

